cmd/run.go

```diff
@@ -330,7 +330,7 @@ func runNonInteractive(ctx context.Context, flags RunFlags) error {
 
 	github.GenerateWorkflowSummary(ctx, workflow.RootStep)
 
-	if flags.LaunchStudio || shouldLaunchStudio(ctx, workflow, false) {
+	if (flags.LaunchStudio || shouldLaunchStudio(ctx, workflow, false)) && isSupportedInputSpecForStudio(ctx, workflow.GetWorkflowFile()) {
```

Comment:
This should be moved inside `shouldLaunchStudio` because quickstart also calls that

Reply:
I also call it in quickstart

Reply:
Additionally it doesn't belong in that method because we have the cli flag override that needs to be respected

Reply:
Still though seems cleaner to be called from `shouldLaunchStudio` but is a nit

Reply:
The logical OR wouldn't make sense then

Reply:
I can move it if you feel strongly though

Reply:
@mfbx9da4 Edited my original comment to explain my reasoning better

Reply:
That's a good point, we also have inside the `shouldLaunchStudio` check for multiple source being run which will throw a hard error later on. So I think we should have

```
if flags.Watch {
  cannotLaunchReason := canLaunchStudio()
  if cannotLaunchReason {
   warn(cannotLaunchReason)
  } else {
     launch()
}
}
```

also now out of date with main btw